File: shell-examples/src/main/js/io/vertx/example/shell/helloworld/hello_world_command.js
var CommandBuilder = require("vertx-shell-js/command_builder");
var ShellService = require("vertx-shell-js/shell_service");
var CommandRegistry = require("vertx-shell-js/command_registry");

var helloWorld = CommandBuilder.command("hello-world").processHandler(function (process) {
  process.write("hello world\n");
  process.end();
}).build(vertx);

var service = ShellService.create(vertx, {
  "telnetOptions" : {
    "host" : "localhost",
    "port" : 3000
  }
});
CommandRegistry.getShared(vertx).registerCommand(helloWorld);
service.start(function (ar, ar_err) {
  if (!ar_err == null) {
    ar_err.printStackTrace();
  }
});

File: shell-examples/src/main/kotlin/io/vertx/example/shell/helloworld/HelloWorldCommand.kt
package io.vertx.example.shell.helloworld

import io.vertx.ext.shell.ShellService
import io.vertx.ext.shell.ShellServiceOptions
import io.vertx.ext.shell.command.CommandBuilder
import io.vertx.ext.shell.command.CommandRegistry
import io.vertx.ext.shell.term.TelnetTermOptions
import io.vertx.kotlin.common.json.*
import io.vertx.kotlin.ext.shell.*
import io.vertx.kotlin.ext.shell.term.*

class HelloWorldCommand : io.vertx.core.AbstractVerticle()  {
  override fun start() {

    var helloWorld = CommandBuilder.command("hello-world").processHandler({ process ->
      process.write("hello world\n")
      process.end()
    }).build(vertx)

    var service = ShellService.create(vertx, ShellServiceOptions(
      telnetOptions = TelnetTermOptions(
        host = "localhost",
        port = 3000)))
    CommandRegistry.getShared(vertx).registerCommand(helloWorld)
    service.start({ ar ->
      if (!ar.succeeded()) {
        ar.cause().printStackTrace()
      }
    })
  }
}
